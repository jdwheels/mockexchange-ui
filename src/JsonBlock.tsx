import React, { FC } from 'react';
import PropTypes from 'prop-types';

interface JsonBlockProps {
  json: Record<string, unknown>;
}

const JsonBlock: FC<JsonBlockProps> = function ({ json }) {
  return (
    <pre>
      <code className="x">
        {JSON.stringify(json, null, 2)}
      </code>
    </pre>
  );
};

JsonBlock.propTypes = {
  json: PropTypes.shape({}).isRequired,
};

export default JsonBlock;
