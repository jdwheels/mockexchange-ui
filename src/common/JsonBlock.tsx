import React, { FC } from 'react';
import PropTypes from 'prop-types';

interface JsonBlockProps {
  json: Record<string, unknown>;
}

const JsonBlock: FC<JsonBlockProps> = function ({ json }) {
  return (
    <pre>
      <code>
        {JSON.stringify(json, null, 2)}
      </code>
    </pre>
  );
};

JsonBlock.propTypes = {
  json: PropTypes.shape({}).isRequired,
};

export default JsonBlock;
