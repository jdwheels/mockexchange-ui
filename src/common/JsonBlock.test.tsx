import React from 'react';
import { render } from '@testing-library/react';
import pretty from 'pretty';
import JsonBlock from './JsonBlock';

test('should render blocks', () => {
  const { container } = render(<JsonBlock json={{ some: 'block' }} />);
  expect(pretty(container.innerHTML)).toMatchSnapshot();
});
