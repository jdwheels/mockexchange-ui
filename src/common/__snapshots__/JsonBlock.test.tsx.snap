// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render blocks 1`] = `
"<pre><code>{
  \\"some\\": \\"block\\"
}</code></pre>"
`;
